src/aura/BGE_ConfigurationWizard/BGE_ConfigurationWizardHelper.js

```diff
@@ -1,1058 +1,611 @@
+        let matchOnDateSelected = donationMatchingRule.indexOf(component.get('v.wizardMetadata.namespacePrefix') + 'donation_date__c') >= 0;
+        component.set('v.wizardMetadata.matchOnDateSelected', matchOnDateSelected);
+    },
 
-            /**
-             * @description Validates the required batchInfo.
-             * @return Boolean validity.
-             */
-            function isValid() {
-                return this.name && this.description;
-            }
+    /******************************** Sort and Group Functions *****************************/
 
-            /**
-             * @description Checks if the specified string is a matched substring of any strings in list.
-             * @return Boolean string matches.
-             */
-            function isStringMatchedInList(theList, theString) {
-                let stringMatches = false;
-                theList.forEach(function(currString) {
-                    if (currString.indexOf(theString) >= 0) {
-                        stringMatches = true;
-                    }
-                });
-                return stringMatches;
-            }
+    /**
+     * @description Gets a flat list of the active fields sorted by order.
+     * @return List of active fields.
+     */
+    getActives: function (component) {
+        let allFields = component.get('v.everyField');
+        let activeFields = [];
+        allFields.forEach(function (currentField) {
+            if (currentField.isActive) {
+                activeFields.push(currentField);
+            }
+        });
+        var sortedActiveFields = this.sortFieldsByOrder(activeFields);
+        return sortedActiveFields;
+    },
 
-            // BatchInfo module public functions and properties
-            return {
-                // record info
-                name: '',
-                id: '',
-                description: '',
-                expectedCount: 0,
-                expectedTotal: 0,
-                recordCount: 0,
-
-                // batch processing settings
-                requireTotalMatch: false,
-                batchProcessSize: 0,
-                runOpportunityRollupsWhileProcessing: false,
-                donationMatchingBehavior: '',
-                donationMatchingClass: '',
-                donationMatchingOptions: [],
-                donationMatchingRule: [],
-                donationDateRange: '',
-                postProcessClass: '',
-                load: load,
-                isValid: isValid,
-                noMatchOnDate: false,
-                onInfoUpdated: _onInfoUpdated
-            }
-        })(this.Event());
+    /**
+     * @description Gets the active fields sorted and grouped by SObject.
+     * @return Map of SObject group to List of related active fields.
+     */
+    getActivesBySObject: function (component) {
+        let activeFields = this.getActives(component);
+        var activesBySObject = this.groupFieldsBySObject(activeFields);
```

Comment:
could be `let` if we want

Reply:
there are a lot of remaining `var`s, but this one sticks out because it's right next to a `let`. maybe another PR for upgrading everything to ES6?